### PR TITLE
Initial version of powershell module for configuring settings

### PR DESCRIPTION
**Pull request by ErikApption**
## Description

This adds a powershell module that can take a settings template and either
- generate a standard appsettings.<env>.json file
- populate variables

Secrets from key vault are written as local secrets in each project.

The template follows the same syntax as app service and has `"@Microsoft.KeyVault(VaultName=<vault>;SecretName=<secret>)"`

Also this includes the following scripts
- `gen-all-settings.ps1` this configures settings for all solutions with a template
- `gen-settings.ps1` in each project can be used separately

## How Has This Been Tested?
- Portal runs with new template
- RP Specflow test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Unit test (new or update to tests)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (change that would cause documentation to be updated)
- [X] Configuration change (change that would affect the configuration of the application)

## Checklist:
- [X] PR is submitted to the correct branch `develop`.
- [X] Code follows the code style of this project.
- [X] Changes are covered by Specflow tests and all new or existing tests are passing.
- [X] Any new or updated translatable strings have been added to the localization files.
- [X] Necessary and relevant documentation has been created or updated.
- [X] There are either no changes to the application's configuration or the necessary changes in the infrastructure repository are included in a separate PR.

### CODE REVIEW

**Review by yjmrobert — APPROVED** [review]
Looks good to me, can you rename the `Datahub.Portal` source file from `sand` to `test` and include the template for the portal please 💝

**Comment by ErikApption** [comment]
@yjmrobert this now includes resourceprovisioner and should cover all dotnet projects in solution

File: Portal/src/Datahub.Portal/gen-settings.ps1 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿$CurrentPath = Split-Path -Parent $MyInvocation.MyCommand.Path
+#get full path from $CurrentPath
+
+Import-Module $CurrentPath/../../../scripts/appsettings.psm1 -Force
+
+Export-AppSettings -SourceFile "template.appsettings.sand.json" -Target AppSettings -EnvironmentName test
```
Comment (yjmrobert) [inline Portal/src/Datahub.Portal/gen-settings.ps1:6]: Should be `template.appsettings.json` no?

**Review by yjmrobert — COMMENTED** [review]
Looks good, just need to confirm the `gen-settings.ps1` for the portal

File: Portal/src/Datahub.Portal/gen-settings.ps1 (hunk 1 above)
Comment (ErikApption) [reply Portal/src/Datahub.Portal/gen-settings.ps1:6]: I haven't fixed the individual scripts. Right now this assumes you run the gen-all-settings.ps1

**Pull request merged**